functions/http/main.py

```diff
@@ -118,3 +118,56 @@ def get_signed_url(request):
 
     return url
 # [END functions_http_signed_url]
+
+
+# [START functions_http_cors]
+def cors_enabled_function(request):
+    # Set CORS headers for the preflight request
+    # e.g. allows GETs from any origin with the Content-Type header
+    # and cache preflight response for an 3600s
+
+    # Send response to OPTIONS requests and terminate the function execution
+    if request.method == 'OPTIONS':
+        headers = {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET',
+            'Access-Control-Allow-Headers': 'Content-Type',
+            'Access-Control-Max-Age': '3600'
+        }
+
+        return ('', 204, headers)
+
+    # Set CORS headers for the main request
```

Comment:
"The main request" is ambiguous here. Maybe we should just say "Set CORS headers" or perhaps "Set CORS headers for non-OPTIONS requests" if that is necessary? Also: same comment on similar line in next function.